index.js

```diff
@@ -7,6 +7,23 @@ import { createSvelteSlots, findSlotParent, unwrap } from './utils.js';
 const propMapCache = new Map();
 
 
+// Mutation observer must be used to track changes to attributes on our custom elements, since we cannot know the
+// component props ahead of time (required if we were to use observedAttributes instead). In this case, only one
+// observer is necessary, since each call to .observe() can have a different "attributeFilter" specified.
+// NOTE: We can .observe() many separate elements and not have to .disconnect() each one individually, since if the
```

Comment:
> and do* not have to

Also take moment to convert this to block style (too many `// …`). Same for `propMapCache` above.